Code/GraphMol/Wrap/MolOps.cpp

```diff
@@ -305,9 +305,15 @@ void addRecursiveQueriesHelper(ROMol &mol, python::dict replDict,
   addRecursiveQueries(mol, replacements, propName);
 }
 
-ROMol *addHs(const ROMol &orig, bool explicitOnly = false,
-             bool addCoords = false) {
-  return MolOps::addHs(orig, explicitOnly, addCoords);
+ROMol *addHs(const ROMol &orig, bool explicitOnly, bool addCoords,
+             python::object onlyOnAtoms) {
+  std::vector<unsigned int> *onlyOn = NULL;
+  if (onlyOnAtoms) {
+    onlyOn = pythonObjectToVect(onlyOnAtoms, orig.getNumAtoms());
+  }
+  ROMol *res = MolOps::addHs(orig, explicitOnly, addCoords, onlyOn);
```

Comment:
It doesn't seem like there's really a need to do the check. If you say you only want Hs added to atoms 35 and 36, but your molecule only has 4 atoms, it's arguable whether or not it's actually an error. It doesn't necessarily make a lot of sense, but that's not worth throwing an exception for.


The python side does the check because that function does the check. It's just there already.